tests/baselines/reference/strictFunctionTypes1.js

```diff
@@ -82,7 +82,7 @@ declare function fs(x: string): void;
 declare function fx(f: (x: "def") => void): void;
 declare const x1: (x: string) => void;
 declare const x2 = "abc";
-declare const x3: string;
+declare const x3: "def" | "abc";

```

Comment:
I understand that there is this hidden widening thing encoded in the type and that it actually might widen when assigned to a mutable declaration. I think though that most people are not aware of this and it's *more* surprising that a type widened in the emitted declaration - people never audit them until something goes wrong.

The current behavior is also surprising in some cases. Let's expand on this very case here:
```ts
declare function f3<T>(obj: T, f1: (x: T) => void, f2: (f: (x: T) => void) => void): T;
declare function fo(x: Object): void;
declare function fx(f: (x: "def") => void): void;

const x3 = f3("abc", fo, fx);  // "abc" | "def"

let other: typeof x3 = x3
other = 'other' // error
```

As we can see we can't assign `'other'` to `other` but yet in the emitted declaration file we can see this:
```ts
declare const x3: string;
declare let other: typeof x3;
```

It means that in practice we can end up with an error locally that wouldn't get reported if the same code would get used through the declaration file. I find that problematic. 

The same could be said about the widening thing because "locally" an assignment would widen but if we change the declaration emit then the equivalent assignment would not widen since the widening behavior can't be encoded in the declaration file. I think though that's a smaller problem and a better tradeoff.

Reply:
> and it's more surprising that a type widened in the emitted declaration - people never audit them until something goes wrong.

Correct! And in most cases in declaration emit, the way we handle this is by preserving the literal initializer, rather than serializing a type annotation - that way the declaration has the same widening behavior as the original source. Eg, if you write

```ts
export const six = 6;
```
that is also, verbatim, the declaration emit. Unfortunately, right now this only covers simple cases of widening types - a heritage from our older, much simpler string-based declaration emitter.

IMO, rather than making an equally incorrect declaration in the other direction (preserving the literal types, rather than the widened type), we need to, once again, preserve the initializer to preserve the literal-widening behavior. In this case, that should mean making initializers with the same behavior.

Take the original issue:
```ts
function foo(): {y: 1} {
    return { y: 1 }
}

export const { y = 0 } = foo();
```

If we serialize it to
```ts
declare const _expr: {y: 1};
export declare const { y = 0 } = _expr;
```
`y` now behaves identically in the declaration file as it does in the original source file. Do note, this means that we'll need to allow identifiers in initializers in declaration files - something currently an error - but I think that's fine, especially since, despite the error, old versions of TS will still typecheck it correctly.